src/clp_ffi_js/ir/StructuredIrStreamReader.hpp

```diff
@@ -0,0 +1,169 @@
+#ifndef CLP_FFI_JS_IR_STRUCTUREDIRSTREAMREADER_HPP
+#define CLP_FFI_JS_IR_STRUCTUREDIRSTREAMREADER_HPP
+
+#include <Array.hpp>
+#include <cstddef>
+#include <memory>
+#include <optional>
+#include <vector>
+
+#include <clp/ffi/ir_stream/Deserializer.hpp>
+#include <clp/ir/LogEventDeserializer.hpp>
+#include <clp/ir/types.hpp>
+#include <clp/TimestampPattern.hpp>
+#include <emscripten/val.h>
+#include <spdlog/spdlog.h>
+
+#include <clp_ffi_js/ir/LogEventWithLevel.hpp>
+#include <clp_ffi_js/ir/StreamReader.hpp>
+#include <clp_ffi_js/ir/StreamReaderDataContext.hpp>
+
+namespace clp_ffi_js::ir {
+using parsed_tree_node_id_t = std::optional<clp::ffi::SchemaTree::Node::id_t>;
+
+/**
+ * Class to handle deserialized IR units.
+ */
+class IrUnitHandler {
```

Comment:
Do we want to move this class into a separate header? That way we don't have to write `using StructuredIrDeserializer = clp::ffi::ir_stream::Deserializer<IrUnitHandler>;` after this class.